src/C50A0.c

```diff
@@ -15,11 +15,32 @@ INCLUDE_ASM(s32, "C50A0", draw_image_with_clipping);
 
 INCLUDE_ASM(s32, "C50A0", draw_tiled_image);
 
-INCLUDE_ASM(s32, "C50A0", integer_log);
+s32 integer_log(s32 number, s32 base) {
+    f64 temp_f0;
+    f32 fNumber = number;
+    s32 ret = 1;
+
+    do {
+        temp_f0 = base;
+        if (base < 0) {
+            temp_f0 += 4294967296.0;
+        }
+        fNumber /= (f32)temp_f0;
+        if (fNumber <= 1.0) {
+            return ret;
+        }
+
+        ret++;
+    } while (TRUE);
```

Comment:
I'd prefer a `while (TRUE) { ... }` construct for an unconditional loop personally